src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.Placement.cs

```diff
@@ -0,0 +1,384 @@
+		new Dictionary<FlyoutBase.MajorPlacementMode, Memory<FlyoutBase.MajorPlacementMode>>()
+		{
+			 {FlyoutBase.MajorPlacementMode.Top, new []
+			 {
+				 FlyoutBase.MajorPlacementMode.Top,
+				 FlyoutBase.MajorPlacementMode.Bottom,
+				 FlyoutBase.MajorPlacementMode.Left,
+				 FlyoutBase.MajorPlacementMode.Right,
+				 FlyoutBase.MajorPlacementMode.Top // use preferred choice if no others fit
+			 }},
+			 {FlyoutBase.MajorPlacementMode.Bottom, new []
+			 {
+				 FlyoutBase.MajorPlacementMode.Bottom,
+				 FlyoutBase.MajorPlacementMode.Top,
+				 FlyoutBase.MajorPlacementMode.Left,
+				 FlyoutBase.MajorPlacementMode.Right,
+				 FlyoutBase.MajorPlacementMode.Bottom // use preferred choice if no others fit
+			 }},
+			 {FlyoutBase.MajorPlacementMode.Left, new []
+			 {
+				 FlyoutBase.MajorPlacementMode.Left,
+				 FlyoutBase.MajorPlacementMode.Right,
+				 FlyoutBase.MajorPlacementMode.Top,
+				 FlyoutBase.MajorPlacementMode.Bottom,
+				 FlyoutBase.MajorPlacementMode.Left // use preferred choice if no others fit
+			 }},
+			 {FlyoutBase.MajorPlacementMode.Right, new []
+			 {
+				 FlyoutBase.MajorPlacementMode.Right,
+				 FlyoutBase.MajorPlacementMode.Left,
+				 FlyoutBase.MajorPlacementMode.Top,
+				 FlyoutBase.MajorPlacementMode.Bottom,
+				 FlyoutBase.MajorPlacementMode.Right // use preferred choice if no others fit
+			 }},
+		};
+
+	/// <summary>
+	/// This value is an arbitrary value for the placement of
+	/// a popup below its anchor.
+	/// </summary>
+	protected virtual int PopupPlacementTargetMargin => 5;	
+
+	private void Window_SizeChanged(object sender, WindowSizeChangedEventArgs e)
+		=> InvalidateMeasure();
+
+	// TODO: Use this whenever popup placement is Auto
+	protected virtual PopupPlacementMode DefaultPopupPlacement { get; }
+
+	protected virtual bool FullPlacementRequested { get; }
+
+	internal virtual FlyoutBase Flyout => null;
+
+	private Size PlacementArrangeOverride(Popup popup, Size finalSize)
+	{
+		foreach (var child in Children)
+		{
+			var desiredSize = child.DesiredSize;
+			var maxSize = (child as FrameworkElement).GetMaxSize(); // UWP takes FlyoutPresenter's MaxHeight and MaxWidth into consideration, but ignores Height and Width
+			var rect = CalculatePopupPlacement(popup, desiredSize, maxSize);
+
+			if (Flyout?.IsTargetPositionSet ?? false)
+			{
+				rect = Flyout.UpdateTargetPosition(ApplicationView.GetForCurrentView().VisibleBounds, desiredSize, rect);
+			}
+
+			child.Arrange(rect);
+		}
+
+		return finalSize;
+	}
+
+	private Rect? GetAnchorRect(Popup popup)
+	{
+#if __ANDROID__ || __IOS__
+		if (NativeAnchor != null)
+		{
+			return NativeAnchor.GetBoundsRectRelativeTo(this);
+		}
+#endif
+		var anchor = popup.PlacementTarget;
+		if (anchor == null)
+		{
+			return default;
+		}
+
+		return anchor.GetBoundsRectRelativeTo(this);
+	}
+
+	protected virtual Rect CalculatePopupPlacement(Popup popup, Size desiredSize, Size maxSize)
```

Comment:
It seems this is only called in this class, consider:

```suggestion
	private Rect CalculatePopupPlacement(Popup popup, Size desiredSize, Size maxSize)
```

Reply:
The method wasn't renamed before, but there is an override for native picker popups in iOS, so need to keep this overridable 👍